src/targets/gpu/include/migraphx/gpu/miopen.hpp

```diff
@@ -104,14 +104,17 @@ inline convolution_descriptor make_deconv(const T& op)
 {
     auto c = make_obj<convolution_descriptor>(&miopenCreateConvolutionDescriptor);
     miopenConvolutionMode_t c_mode = miopenTranspose;
-    miopenInitConvolutionDescriptor(c.get(),
-                                    c_mode,
-                                    op.padding[0],
-                                    op.padding[1],
-                                    op.stride[0],
-                                    op.stride[1],
-                                    op.dilation[0],
-                                    op.dilation[1]);
+    int kdims                      = op.kdims();
+    std::vector<int> padding(std::max(2, kdims), 0);
+    std::vector<int> stride(std::max(2, kdims), 1);
+    std::vector<int> dilation(std::max(2, kdims), 1);
+
```

Comment:
Can we add a assert here to ensure padding.size() == op.padding.size(), same for stride and dilation.

Reply:
it may not to be equal if the op is 1d, so this check is not necessary. However, when I call op.kdims(), there is an assert to make sure the attributes are consistent.